src/lib/Libifl/pbsD_msgjob.c

```diff
@@ -202,6 +202,50 @@ char *extend;
 	if (pbs_client_thread_init_thread_context() != 0)
 		return pbs_errno;
 
+	/* first verify the resource list in keep_select option */
+	if (extend) {
+		struct attrl *attrib = NULL;
+		char ebuff[200], *erp, *emsg = NULL;
+		int i;
+		struct pbs_client_thread_connect_context *con;
+		char nd_ct_selstr[20];
+		char *endptr = NULL;
+
+		strtol(extend, &endptr, 10);
```

Comment:
You should check the return value and print strerror(errno) if the conversion fails.

Reply:
I thought of this, but since I'm using this `strtol()` to detect the case where the argument to `-k` option is just numbers and not a select spec, I feel the checks and the error logs will be intrusive with the else case and might confuse the user.

Reply:
I think I could break this code using: -k '123    '
*endptr would not point to '\0' in this case.

Reply:
Please try after the new commit https://github.com/PBSPro/pbspro/pull/1452/commits/b33b0c1672d0a8fc21ece2ce23438681e42db09f